New Resource: `azurerm_cosmosdb_mongo_collection` (#3459)
terraform-providers/terraform-provider-azurerm: website/docs/r/cosmosdb_mongo_collection.html.markdown

  
  default_ttl_seconds = "777"   
  shard_key           = "uniqueKey"
  
  indexes {
    key    = "aKey"
    unique = false
  }
  
  indexes {
    key    = "uniqueKey"
    unique = true
  }
}

```

## Argument Reference

The following arguments are supported:

* `name` - (Required) Specifies the name of the Cosmos DB Mongo Collection. Changing this forces a new resource to be created.
* `resource_group_name` - (Required) The name of the resource group in which the Cosmos DB Mongo Collection is created. Changing this forces a new resource to be created.
* `resource_group_name` - (Required) The name of the Cosmos DB Mongo Database in which the Cosmos DB Mongo Collection is created. Changing this forces a new resource to be created.
* `default_ttl_seconds` - (Required) The default Time To Live in seconds. If the value is `-1` items are not automatically expired.
* `shard_key` - (Required) The name of the key to partition on for sharding. There must not be any other unique index keys. 
* `indexes` - (Optional) One or more `indexes` blocks as defined below.

---

An `indexes` block supports the following:

* `key` - (Required) The name of the key to use for this index.

> * `unique` - (Required) Should the index key be unique?

Reviewer: I've always found phrasing the description as a question to be awkward. Do you think 
```suggestion
* `unique` - (Required) Whether the index key should be unique.
```
sounds a little better?

## Attributes Reference

The following attributes are exported:

> * `id` - the Cosmos DB Mongo Database ID.

Reviewer: ```suggestion
* `id` - the Cosmos DB Mongo Collection ID.
```

## Import

> Cosmos Mongo KeySpace can be imported using the `resource id`, e.g.

Reviewer: ```suggestion
Cosmos DB Mongo Collection can be imported using the `resource id`, e.g.
```
